pkg/cachedmap/cachedmap_test.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cachedmap
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const defaultTestTTL = 2 * time.Second
+
+func TestNewCachedMap(t *testing.T) {
+	t.Parallel()
+
+	cache := NewCachedMap[int, string](defaultTestTTL).(*cachedMap[int, string])
+	require.NotNil(t, cache)
+	defer cache.Close()
```

Comment:
nit: AFAIU it's a better practice to use test.Cleanup() instead of defers in tests. (https://stackoverflow.com/questions/61609085/what-is-useful-for-t-cleanup)